src/test/kotlin/sirius/kernel/commons/ValueTest.kt

```diff
@@ -0,0 +1,246 @@
+        assertEquals(99999, Value.of("99999").asInt(1234))
+        assertEquals(1234, Value.of("Keine Zahl").asInt(1234))
+        assertEquals(1234, Value.of(null).asInt(1234))
+    }
+
+    @Test
+    fun `Casting to integer instances works`() {
+        val boxedInt: Int? = 99999
+        assertEquals(boxedInt, Value.of(99999).getInteger())
+        assertEquals(boxedInt, Value.of(99999L).getInteger())
+        assertEquals(boxedInt, Value.of(99999.0).getInteger())
+        assertEquals(boxedInt, Value.of((99999).toDouble()).getInteger())
+        assertEquals(boxedInt, Value.of(Amount.of((99999).toInt())).getInteger())
+        assertEquals(null, Value.of(Amount.NOTHING).getInteger())
+        assertEquals(boxedInt, Value.of("99999").getInteger())
+        assertEquals(null, Value.of("Keine Zahl").getInteger())
+        assertEquals(null, Value.of(null).getInteger())
+    }
+
+    @Test
+    fun `Casting to longs works`() {
+        assertEquals(99999L, Value.of(99999).asLong(1234))
+        assertEquals(99999L, Value.of(99999L).asLong(1234))
+        assertEquals(99999L, Value.of(99999.0).asLong(1234))
+        assertEquals(99999L, Value.of((99999).toDouble()).asLong(1234))
+        assertEquals(99999L, Value.of(Amount.of((99999).toInt())).asLong(1234))
+        assertEquals(1234L, Value.of(Amount.NOTHING).asLong(1234))
+        assertEquals(99999L, Value.of("99999").asLong(1234))
+        assertEquals(1234L, Value.of("Keine Zahl").asLong(1234))
+        assertEquals(1234L, Value.of(null).asLong(1234))
+    }
+
+    @Test
+    fun `Casting to long instances works`() {
+        val boxedLong: Long? = 99999
+        assertEquals(boxedLong, Value.of(99999).getLong())
+        assertEquals(boxedLong, Value.of(99999L).getLong())
+        assertEquals(boxedLong, Value.of(99999.0).getLong())
+        assertEquals(boxedLong, Value.of((99999).toDouble()).getLong())
+        assertEquals(boxedLong, Value.of(Amount.of((99999).toInt())).getLong())
+        assertEquals(null, Value.of(Amount.NOTHING).getLong())
+        assertEquals(boxedLong, Value.of("99999").getLong())
+        assertEquals(null, Value.of("Keine Zahl").getLong())
+        assertEquals(null, Value.of(null).getLong())
+    }
+
+    @Test
+    fun `map() does not call the mapper on an empty Value`() {
+        var count = 0
+        val mapper: (value: Any) -> String = { value ->
+            count++
+            ""
+        }
+        Value.EMPTY.map(mapper)
+        assertEquals(0, count)
+    }
+
+    @Test
+    fun `flatMap() does not call the mapper on an empty Value`() {
+        var count = 0
+        val mapper: (value: Any) -> Optional<Value> = { value ->
+            count++
+            Optional.empty()
+        }
+        Value.EMPTY.flatMap(mapper)
+        assertEquals(0, count)
+    }
+
+    @Test
+    fun `asOptionalInt must not throw NPE on floats`() {
+        val value = Value.of(1.1f)
+        assertDoesNotThrow {
```

Comment:
I would remove assertDoesNotThrow. Not required here (was introduced to follow spock conventions)